src/Chapter3.hs

```diff
@@ -971,6 +1040,20 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Mon | Tue | Wed | Thu | Fri | Sat | Sun deriving (Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend Sat = True
+isWeekend Sun = True
+isWeekend _ = False
+
+nextDay :: Weekday -> Weekday
+nextDay = succ
```

Comment:
Unfortunately, this implementation is not exactly correctly, because `succ` fails in runtime on the last element of enumeration instead of returning the first element. You can verify this in GHCi:

```haskell
ghci> succ Sun
*** Exception: succ: bad argument
```